gemrb/core/Map.cpp

```diff
@@ -787,17 +783,33 @@ void Map::UpdateScripts()
 		actor->fxqueue.Cleanup();
 	}
 
-	// We need to step through the list of actors until all of them are done
-	// taking steps.
+	// We need to step through the list of actors until
+	// all of them are done taking steps
 	bool more_steps = true;
 	ieDword time = game->Ticks; // make sure everything moves at the same time
 	while (more_steps) {
 		more_steps = false;
-
+		// Make all actors pathfind if there are others nearby
+		// in order to avoid bumping when possible
 		q=Qcount[PR_SCRIPT];
 		while (q--) {
 			Actor* actor = queue[PR_SCRIPT][q];
-			more_steps = !DoStepForActor(actor, actor->speed, time);
+			if (actor->GetRandomBackoff()) {
+				continue;
+			}
+			Actor* nearActor = GetActorInRadius(actor->Pos, GA_NO_DEAD|GA_NO_UNSCHEDULED, actor->GetAnims()->GetCircleSize());
+			if (nearActor && nearActor != actor) {
+				actor->NewPath();
+			}
+		}
+		q=Qcount[PR_SCRIPT];
```

Comment:
I know you just copied it, but this is definitely one of the future style rules — spaces around operands (already checked for python PRs):
```suggestion
		q = Qcount[PR_SCRIPT];
```